src/Region/Region.cc

```diff
@@ -1311,3 +1356,55 @@ bool Region::ConvertWorldToPixel(std::vector<casacore::Quantity>& world_point, s
 std::shared_mutex& Region::GetActiveTaskMutex() {
     return _active_task_mutex;
 }
+
+void Region::RemoveHorizontalPolygonPoints(casacore::Vector<casacore::Double>& x, casacore::Vector<casacore::Double>& y) {
+    // When polygon points have close y-points (horizontal segment), the x-range is masked only to the next point.
+    // Remove points not near integral pixel.
+    std::vector<casacore::Double> keep_x, keep_y;
+    size_t npoints(x.size());
+
+    for (int i = 0; i < npoints - 2; ++i) {
+        if (i == 0) {
+            // always include first point of segment
+            keep_x.push_back(x[i]);
+            keep_y.push_back(y[i]);
+            continue;
+        }
+
+        float this_y = y[i];
+        float next_y = y[i + 1];
+        if (!ValuesNear(this_y, next_y)) {
+            // Line connecting points not ~horizontal - keep point
+            keep_x.push_back(x[i]);
+            keep_y.push_back(y[i]);
+            continue;
+        }
+
+        // Line connecting points ~horizontal - keep point nearest integral pixel
+        int pixel_y = static_cast<int>(this_y);
+
+        if (!ValuesNear(this_y, float(pixel_y))) {
+            // Skip point not near pixel
+            continue;
+        }
+
+        if ((static_cast<int>(next_y) == pixel_y) && ((this_y - pixel_y) > (next_y - pixel_y))) {
```

Comment:
I am just wondering is it better using `fabs(this_y - pixel_y) > fabs(next_y - pixel_y)` than `(this_y - pixel_y) > (next_y - pixel_y)`?

Reply:
pixel_y is the integer part of the float values y and next _y, so these subtraction values will always be positive.